container/podman/podman.go

```diff
@@ -0,0 +1,161 @@
+	}
+	return StatusFromPodmanInfo(podmanInfo)
+}
+
+func StatusFromPodmanInfo(podmanInfo dockertypes.Info) (v1.PodmanStatus, error) {
+	out := v1.PodmanStatus{}
+	out.KernelVersion = machine.KernelVersion()
+	out.OS = podmanInfo.OperatingSystem
+	out.Hostname = podmanInfo.Name
+	out.RootDir = podmanInfo.DockerRootDir
+	out.Driver = podmanInfo.Driver
+	out.NumImages = podmanInfo.Images
+	out.NumContainers = podmanInfo.Containers
+	out.DriverStatus = make(map[string]string, len(podmanInfo.DriverStatus))
+	for _, v := range podmanInfo.DriverStatus {
+		out.DriverStatus[v[0]] = v[1]
+	}
+	var err error
+	ver, err := VersionString()
+	if err != nil {
+		return out, err
+	}
+	out.Version = ver
+	ver, err = APIVersionString()
+	if err != nil {
+		return out, err
+	}
+	out.APIVersion = ver
+	return out, nil
+}
+
+// Checks whether the podmanInfo reflects a valid docker setup, and returns it if it does, or an
+// error otherwise.
+func ValidateInfo() (*dockertypes.Info, error) {
+	client, err := Client()
+	if err != nil {
+		return nil, fmt.Errorf("unable to communicate with podman: %v", err)
+	}
+
+	podmanInfo, err := client.Info(defaultContext())
+	if err != nil {
+		return nil, fmt.Errorf("failed to detect podman info: %v", err)
+	}
+
+	// Fall back to version API if ServerVersion is not set in info.
+	if podmanInfo.ServerVersion == "" {
+		version, err := client.ServerVersion(defaultContext())
+		if err != nil {
+			return nil, fmt.Errorf("unable to get podman version: %v", err)
+		}
+		podmanInfo.ServerVersion = version.Version
+	}
+
+	if podmanInfo.Driver == "" {
+		return nil, fmt.Errorf("failed to find podman storage driver")
+	}
+
+	return &podmanInfo, nil
+}
+
+func APIVersion() ([]int, error) {
+	ver, err := APIVersionString()
+	if err != nil {
+		return nil, err
+	}
+	return parseVersion(ver, apiVersionRe, 2)
+}
+
+func VersionString() (string, error) {
+	podmanVersion := "Unknown"
+	client, err := Client()
+	if err == nil {
+		version, err := client.ServerVersion(defaultContext())
+		if err == nil {
+			podmanVersion = version.Version
+		}
+	}
+	return podmanVersion, err
+}
+
+func APIVersionString() (string, error) {
+	apiVersion := "Unknown"
+	client, err := Client()
+	if err == nil {
+		version, err := client.ServerVersion(defaultContext())
+		if err == nil {
+			apiVersion = version.APIVersion
+		}
+	}
+	return apiVersion, err
+}
+
+func parseVersion(versionString string, regex *regexp.Regexp, length int) ([]int, error) {
+	matches := regex.FindAllStringSubmatch(versionString, -1)
+	if len(matches) != 1 {
+		return nil, fmt.Errorf("version string \"%v\" doesn't match expected regular expression: \"%v\"", versionString, regex.String())
+	}
+	versionStringArray := matches[0][1:]
+	versionArray := make([]int, length)
+	for index, versionStr := range versionStringArray {
+		version, err := strconv.Atoi(versionStr)
+		if err != nil {
+			return nil, fmt.Errorf("error while parsing \"%v\" in \"%v\"", versionStr, versionString)
```

Comment:
could it be %q instead of \"%v\"?

Reply:
It's copied from docker/docker.go and boils IMHO down to a "\d+" match and thus could be multi-char string in case of a version segment greater "9". docker_test.go checks for this case and thus i think it's ok. 

Refactoring this version parser stuff into a separate driver independent package would be a good idea.